lib/IRGen/GenBuiltin.cpp

```diff
@@ -711,6 +711,9 @@ if (Builtin.ID == BuiltinValueKind::id) { \
     return;
   }
 
+  if (Builtin.ID == BuiltinValueKind::Ifdef) {
+    llvm_unreachable("Ifdef not constant folded");
```

Comment:
nitpick: I'd prefer a error message describing the root cause: "Builtin.ifdef can only be used in @_alwaysEmitIntoClient functions"